src/BridgedRoom.ts

```diff
@@ -657,6 +657,102 @@ export class BridgedRoom {
         this.dirty = true;
     }
 
+    private async handleSlackMessageFile(file: ISlackFile, slackEventId: string, ghost: SlackGhost) {
+        const maxUploadSize = this.main.config.homeserver.max_upload_size;
+        const channelId = this.slackChannelId;
+        if (!channelId) {
+            // The ID is required.
+            return;
+        }
+
+        if (file.mode === "snippet") {
+            let htmlString: string;
+            try {
+                htmlString = await rp({
+                    headers: {
+                        Authorization: `Bearer ${this.SlackClient!.token}`,
+                    },
+                    uri: file.url_private!,
+                });
+            } catch (ex) {
+                log.error("Failed to download snippet", ex);
+                return;
+            }
+            let htmlCode = "";
+            // Because escaping 6 backticks is not good for readability.
+            // tslint:disable-next-line: prefer-template
+            const code = "```" + `\n${htmlString}\n` + "```";
+            if (file.filetype) {
+                htmlCode = `<pre><code class="language-${file.filetype}'">`;
+            } else {
+                htmlCode = "<pre><code>";
+            }
+            htmlCode += substitutions.htmlEscape(htmlString);
+            htmlCode += "</code></pre>";
+
+            const messageContent = {
+                body: code,
+                format: "org.matrix.custom.html",
+                formatted_body: htmlCode,
+                msgtype: "m.text",
+            };
+            await ghost.sendMessage(this.matrixRoomId, messageContent, channelId, slackEventId);
```

Comment:
We don't seem to check or handle a possible rejection due to the snippet possibly being too long.
I'm not asking you to implement this now, but file it, if you think that's a valid feature request that will come up.

Reply:
Good point, we should upload as a file if it's more than >~64Kb

Reply:
Created issue #450 to track.

Reply:
Opened #460